doc/development/adding_operators.rst

```diff
@@ -0,0 +1,356 @@
+       DeviceArray(0.3, dtype=float32, weak_type=True)]
+
+#. **Non-trainable hyperparameters** that influence the action of the operator. Not every operator has hyperparameters.
+
+   .. code-block:: python
+
+       >>> op.hyperparameters
+       {}
+
+#. Possible **symbolic or numerical representations** of the operator, which can be used by PennyLane's
+   devices to interpret the map. Examples are:
+
+   * Representation as a **product of operators**:
+
+     .. code-block:: python
+
+         >>> op = qml.Rot(0.1, 0.2, 0.3, wires=["a"])
+         >>> op.decomposition()
+         [RZ(0.1, wires=['a']), RY(0.2, wires=['a']), RZ(0.3, wires=['a'])]
+
+   * Representation as a **linear combination of operators**:
+
+     .. code-block:: python
+
+         >>> op = qml.Hamiltonian([1., 2.], [qml.PauliX(0), qml.PauliZ(0)])
+         >>> op.terms()
+         ((1.0, 2.0), [PauliX(wires=[0]), PauliZ(wires=[0])])
+
+   * Representation via the **eigenvalue decomposition** specified by eigenvalues (for the diagonal matrix)
+     and diagonalizing gates (for the unitaries):
+
+     .. code-block:: python
+
+         >>> op = qml.PauliX(0)
+         >>> op.diagonalizing_gates()
+         [Hadamard(wires=[0])]
+         >>> op.eigvals()
+         [ 1 -1]
+
+   * Representation as a **matrix**, as specified by a global wire order that tells us where the
+     wires are found on a register:
+
+     .. code-block:: python
+
+         >>> op = qml.PauliRot(0.2, "X", wires=["b"])
+         >>> op.matrix(wire_order=["a", "b"])
+         [[9.95e-01-2.26e-18j 2.72e-17-9.98e-02j, 0+0j, 0+0j]
+          [2.72e-17-9.98e-02j 9.95e-01-2.26e-18j, 0+0j, 0+0j]
+          [0+0j, 0+0j, 9.95e-01-2.26e-18j 2.72e-17-9.98e-02j]
+          [0+0j, 0+0j, 2.72e-17-9.98e-02j 9.95e-01-2.26e-18j]]
+
+   * Representation as a **sparse matrix**:
+
+     .. code-block:: python
+
+         >>> from scipy.sparse.coo import coo_matrix
+         >>> row = np.array([0, 1])
+         >>> col = np.array([1, 0])
+         >>> data = np.array([1, -1])
+         >>> mat = coo_matrix((data, (row, col)), shape=(4, 4))
+         >>> op = qml.SparseHamiltonian(mat, wires=["a"])
+         >>> op.sparse_matrix(wire_order=["a"])
+         (0, 1)   1
+         (1, 0) - 1
+
+New operators can be created by applying arithmetic functions to operators, such as addition, scalar multiplication,
+multiplication, taking the adjoint, or controlling an operator. At the moment, such arithmetic is only implemented for
+specific subclasses.
+
+* Operators inheriting from :class:`~.Observable` support addition and scalar multiplication:
+
+  .. code-block:: pycon
+
+      >>> op = qml.PauliX(0) + 0.1 * qml.PauliZ(0)
+      >>> op.name
+      Hamiltonian
+      >>> op
+        (0.1) [Z0]
+      + (1.0) [X0]
+
+* Operators inheriting from :class:`~.Operation` define a hermitian conjugate:
+
+  .. code-block:: pycon
+
+      >>> qml.RX(1., wires=0).adjoint()
+      RX(-1.0, wires=[0])
```

Comment:
Wait, they do? I thought this had to be defined on a case by case basis?

Also, I wonder if we shouldn't show `.adjoint()` here? Since it could cause confusion vs. `qml.adjoint(op)`

Reply:
Oh yes, they MAY define...will change this.

I think in this dev context it is useful, because it matches the other representation methods...